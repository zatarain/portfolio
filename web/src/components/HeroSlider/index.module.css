.hero {
	margin: 0 auto;
	text-align: center;
	position: relative;
	overflow: hidden;
	top: -2rem;
	max-width: 1440px;
	max-height: 1080px;
}

.hero::before {
	content: '';
	position: absolute;
	z-index: 2;
	height: 100%;
	bottom: 0;
	left: 0;
	right: 0;
	box-shadow: 0 -7rem 7rem 7rem var(--bg-primary) inset;
	clip-path: ellipse(100% 50% at center 42.6%);
}

.hero img {
	display: inline-block;
	position: relative;
	clip-path: ellipse(100% 50% at center 42.5%);
}

.hero figure {
  margin: 0;
}

.hero figcaption {
  display: none;
}

.hero ul.control-dots {
	bottom: 2rem;
	z-index: 9;
}

.hero ul.control-dots li {
  width: 0.5rem;
  height: 0.5rem;
  padding: 0.5rem;
}
