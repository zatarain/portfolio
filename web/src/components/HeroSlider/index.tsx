import 'react-responsive-carousel/lib/styles/carousel.min.css';
import { Carousel } from 'react-responsive-carousel';
import Image from 'next/image';

interface Properties {
	images: object
}

const HeroSlider = ({ images }: Properties) => {
	return (
		<div className="hero">
			<Carousel
				animationHandler="fade"
				autoPlay={true}
				showThumbs={false}
				showStatus={false}
				transitionTime={1000}
				interval={4000}
				showArrows={false}
				infiniteLoop={true}
				stopOnHover={true}
				swipeable={false}
			>
				{
					Object.entries(images).map(([key, value]) => (
						<div key={key}>
							<Image src={value} alt={key} width={1440} height={1080} />
						</div>
					))
				}
			</Carousel>
		</div>
	)
}

export default HeroSlider
