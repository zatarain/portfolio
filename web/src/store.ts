import { configureStore, ThunkAction, Action, createSlice } from '@reduxjs/toolkit'

export function makeStore() {
  const responsiveFlipper = createSlice({
    name: 'flipper',
    initialState: {
      responsive: false,
    },
    reducers: {
      flipResponsive: (state) => {
        state.responsive = !state.responsive
      }
    }
  })
  return configureStore({
    reducer: { flipper: responsiveFlipper.reducer },
  })
}

const store = makeStore()

export type AppState = ReturnType<typeof store.getState>

export type AppDispatch = typeof store.dispatch

export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  AppState,
  unknown,
  Action<string>
>

export default store
