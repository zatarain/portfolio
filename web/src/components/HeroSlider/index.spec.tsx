import { Provider } from 'react-redux'
import store from '#store'
import { render } from '@testing-library/react'
import HeroSlider from '.'

describe('<HeroSlider />', () => {
	it('renders the component correctly with passed name', () => {
		const pictures = [
			{
				id: '1001',
				media_url: 'https://cdn.instagram.com/dummy-01.jpg',
				caption: 'Dummy caption 01',
			},
			{
				id: '1001',
				media_url: 'https://cdn.instagram.com/dummy-01.jpg',
				caption: 'Dummy caption 01',
			},
		]

		const { asFragment } = render(
			<Provider store={store}>
				<HeroSlider images={pictures} />
			</Provider>
		)

		expect(asFragment()).toMatchSnapshot()
	})
})
