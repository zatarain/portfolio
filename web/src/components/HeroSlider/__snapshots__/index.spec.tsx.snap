// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeroSlider /> renders the component correctly with passed name 1`] = `
<DocumentFragment>
  <div
    class="carousel-root hero"
  >
    <div
      class="carousel carousel-slider"
      style="width: 100%;"
    >
      <ul
        class="control-dots"
      >
        <li
          aria-label="slide item 1"
          class="dot selected"
          role="button"
          tabindex="0"
          value="0"
        />
        <li
          aria-label="slide item 2"
          class="dot"
          role="button"
          tabindex="0"
          value="1"
        />
      </ul>
      <button
        aria-label="previous slide / item"
        class="control-arrow control-prev control-disabled"
        type="button"
      />
      <div
        class="slider-wrapper axis-horizontal"
      >
        <ul
          class="slider animated"
        >
          <li
            class="slide"
            style="position: absolute; display: block; z-index: -2; min-height: 100%; opacity: 0; top: 0px; right: 0px; left: 0px; bottom: 0px; transition-timing-function: ease-in-out; transition-duration: 1000ms;"
          >
            <figure>
              <img
                alt="Dummy caption 01"
                data-nimg="1"
                decoding="async"
                height="768"
                loading="lazy"
                src="/_next/image?url=https%3A%2F%2Fcdn.instagram.com%2Fdummy-01.jpg&w=3840&q=75"
                srcset="/_next/image?url=https%3A%2F%2Fcdn.instagram.com%2Fdummy-01.jpg&w=1920&q=75 1x, /_next/image?url=https%3A%2F%2Fcdn.instagram.com%2Fdummy-01.jpg&w=3840&q=75 2x"
                style="color: transparent;"
                width="1440"
              />
              <figcaption>
                Dummy caption 01
              </figcaption>
            </figure>
          </li>
          <li
            class="slide selected previous"
            style="position: relative; display: block; z-index: -2; min-height: 100%; opacity: 1; top: 0px; right: 0px; left: 0px; bottom: 0px; transition-timing-function: ease-in-out; transition-duration: 1000ms;"
          >
            <figure>
              <img
                alt="Dummy caption 01"
                data-nimg="1"
                decoding="async"
                height="768"
                loading="lazy"
                src="/_next/image?url=https%3A%2F%2Fcdn.instagram.com%2Fdummy-01.jpg&w=3840&q=75"
                srcset="/_next/image?url=https%3A%2F%2Fcdn.instagram.com%2Fdummy-01.jpg&w=1920&q=75 1x, /_next/image?url=https%3A%2F%2Fcdn.instagram.com%2Fdummy-01.jpg&w=3840&q=75 2x"
                style="color: transparent;"
                width="1440"
              />
              <figcaption>
                Dummy caption 01
              </figcaption>
            </figure>
          </li>
          <li
            class="slide"
            style="position: absolute; display: block; z-index: -2; min-height: 100%; opacity: 0; top: 0px; right: 0px; left: 0px; bottom: 0px; transition-timing-function: ease-in-out; transition-duration: 1000ms;"
          >
            <figure>
              <img
                alt="Dummy caption 01"
                data-nimg="1"
                decoding="async"
                height="768"
                loading="lazy"
                src="/_next/image?url=https%3A%2F%2Fcdn.instagram.com%2Fdummy-01.jpg&w=3840&q=75"
                srcset="/_next/image?url=https%3A%2F%2Fcdn.instagram.com%2Fdummy-01.jpg&w=1920&q=75 1x, /_next/image?url=https%3A%2F%2Fcdn.instagram.com%2Fdummy-01.jpg&w=3840&q=75 2x"
                style="color: transparent;"
                width="1440"
              />
              <figcaption>
                Dummy caption 01
              </figcaption>
            </figure>
          </li>
          <li
            class="slide selected previous"
            style="position: relative; display: block; z-index: -2; min-height: 100%; opacity: 1; top: 0px; right: 0px; left: 0px; bottom: 0px; transition-timing-function: ease-in-out; transition-duration: 1000ms;"
          >
            <figure>
              <img
                alt="Dummy caption 01"
                data-nimg="1"
                decoding="async"
                height="768"
                loading="lazy"
                src="/_next/image?url=https%3A%2F%2Fcdn.instagram.com%2Fdummy-01.jpg&w=3840&q=75"
                srcset="/_next/image?url=https%3A%2F%2Fcdn.instagram.com%2Fdummy-01.jpg&w=1920&q=75 1x, /_next/image?url=https%3A%2F%2Fcdn.instagram.com%2Fdummy-01.jpg&w=3840&q=75 2x"
                style="color: transparent;"
                width="1440"
              />
              <figcaption>
                Dummy caption 01
              </figcaption>
            </figure>
          </li>
        </ul>
      </div>
      <button
        aria-label="next slide / item"
        class="control-arrow control-next control-disabled"
        type="button"
      />
    </div>
  </div>
</DocumentFragment>
`;
